src/Uno.UI/NativeFramePresenter.Android.cs

```diff
@@ -160,7 +174,7 @@ private async Task UpdateStack(Page newPage, NavigationTransitionInfo transition
 				var pagesStillInHistory = _frame.BackStack.Select(entry => entry.Instance).ToHashSet();
 				pagesStillInHistory.AddRange(_frame.ForwardStack.Select(entry => entry.Instance));
 				pagesStillInHistory.Add(newPage);
-				_pageStack.Children.Remove(element => !pagesStillInHistory.Contains(element));
+				Children.Remove(element => !pagesStillInHistory.Contains(element));
```

Comment:
Can you add a runtime test that will validate that the pages are not kept in memory when navigating in the right conditions?